plugins/jvm-example/src/main/java/org/elasticsearch/plugin/example/JvmExamplePlugin.java

```diff
@@ -111,5 +117,16 @@ public Settings additionalSettings() {
     public void onModule(RepositoriesModule repositoriesModule) {
     }
 
-
+    /**
+     * Module decalaring some example configuration and a _cat action that uses
+     * it.
+     */
+    public static class ConfiguredExampleModule extends AbstractModule {
+        @Override
+        protected void configure() {
+          bind(ExamplePluginConfiguration.class).asEagerSingleton();
+          Multibinder<AbstractCatAction> catActionMultibinder = Multibinder.newSetBinder(binder(), AbstractCatAction.class);
```

Comment:
It's interesting that this is the way we add cat actions...I wonder if this should be explicit registration like we have for other extension points, rather than having to know to setup a multibinder and what class to bind to. Not relevant for this PR, but I think it would be a good issue to spinoff?